packages/squiggle-lang/src/utility/compactTreeString.ts

```diff
@@ -0,0 +1,77 @@
+import { makeAndToString } from "./FormattedNumber.js";
+import { ImmutableMap } from "./immutable.js";
+
+function formatNumber(number: number): string {
+  return makeAndToString(number, { precision: 3, forceScientific: true });
+}
+
+function formatNumberArray(numbers: number[]): string[] {
+  return numbers.map(formatNumber);
+}
+
+// Converts a tree structure into a compact string representation.
+// Useful for debugging, including sending to LLMs.
+// T can be SimpleValue or SimpleValueWithoutLambda
+export function compactTreeString<T>(
```

Comment:
In `cli/utils.ts`, there's a function called `coloredJson`, with some functionality overlap.

It'd be good to unify this function and `coloredJson`. Compactification would be useful for CLI debugging, not just for LLMs.

Reply:
I'll flag this as a todo for later. ColoredJson takes in any JSON, meant for the AST it looks, while this compactTextString is really for the Value representation.